packages/rs-drive-abci/src/execution/processor.rs

```diff
@@ -0,0 +1,259 @@
+                        .map_err(Error::Drive)?;
+
+                    let balance_change =
+                        individual_fee_result.into_balance_change(identity.id.to_buffer());
+
+                    let outcome = self.drive.apply_balance_change_from_fee_to_identity(
+                        balance_change,
+                        Some(transaction),
+                    )?;
+
+                    // println!("State transition fees {:#?}", outcome.actual_fee_paid);
+                    //
+                    // println!(
+                    //     "Identity balance {:?} changed {:#?}",
+                    //     identity.balance,
+                    //     balance_change.change()
+                    // );
+
+                    Ok(SuccessfulPaidExecution(
+                        validation_result.into_data()?,
+                        outcome.actual_fee_paid,
+                    ))
+                } else {
+                    Ok(ConsensusExecutionError(
+                        SimpleConsensusValidationResult::new_with_errors(validation_result.errors),
+                    ))
+                }
+            }
+            ExecutionEvent::FreeDriveEvent { operations } => {
+                self.drive
+                    .apply_drive_operations(operations, true, block_info, Some(transaction))
+                    .map_err(Error::Drive)?;
+                Ok(SuccessfulFreeExecution)
+            }
+        }
+    }
+
+    /// Processes the given raw state transitions based on the `block_info` and `transaction`.
+    ///
+    /// This function takes a reference to a vector of raw state transitions, `BlockInfo`, and a `Transaction`
+    /// as input and performs the corresponding state transition operations. It deserializes the raw state
+    /// transitions into a `StateTransition` and processes them.
+    ///
+    /// # Arguments
+    ///
+    /// * `raw_state_transitions` - A reference to a vector of raw state transitions.
+    /// * `block_info` - Information about the current block being processed.
+    /// * `transaction` - The transaction associated with the raw state transitions.
+    ///
+    /// # Returns
+    ///
+    /// * `Result<(FeeResult, Vec<ExecTxResult>), Error>` - If the processing is successful, it returns
+    ///   a tuple consisting of a `FeeResult` and a vector of `ExecTxResult`. If the processing fails,
+    ///   it returns an `Error`.
+    ///
+    /// # Errors
+    ///
+    /// This function may return an `Error` variant if there is a problem with deserializing the raw
+    /// state transitions, processing state transitions, or executing events.
+    ///
+    pub(crate) fn process_raw_state_transitions(
+        &self,
+        raw_state_transitions: &Vec<Vec<u8>>,
+        block_info: &BlockInfo,
+        transaction: &Transaction,
+    ) -> Result<(FeeResult, Vec<(Vec<u8>, ExecTxResult)>), Error> {
+        let state_transitions = StateTransition::deserialize_many(raw_state_transitions)?;
```

Comment:
We need to validate against schema before to provide meaningful consenus validation errors